### PR TITLE
xneur: fix build with gcc10

### PR DESCRIPTION
**Pull request by jnetod**
###### Motivation for this change

ZHF: #122042

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 122154` at a6a4a784 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/8d72611f45b9bee1ed6ab93c7eeccf9f">gxneur</a></li>
    <li><a href="https://gist.github.com/6b6fd67987b5b78246d6f2940b59d3c5">xneur</a></li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md">missing-patch-comment</a>

Consider adding a comment explaining the purpose of this patch on the line preceeding.
Near pkgs/applications/misc/xneur/default.nix:25:5:
```
   |
25 |     (fetchpatch {
   |     ^
```
</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

**Comment by jonringer** [comment]
@jnetod thanks for opening your first PR for nixpkgs :)

File: pkgs/applications/misc/xneur/default.nix [hunk 1]
```diff
@@ -21,6 +21,14 @@ stdenv.mkDerivation {
     gst_all_1.gst-plugins-base gst_all_1.gst-plugins-good
   ];
 
+  patches = [
+    (fetchpatch {
+      name = "gcc-10.patch";
+      url = "https://salsa.debian.org/debian/xneur/-/raw/da38ad9c8e1bf4e349f5ed4ad909f810fdea44c9/debian/patches/gcc-10.patch";
+      sha256 = "0sgwb6iab5i9krar22g90rm294c3a99hlnszyr5gca9fbqqcrb95";
```
Comment (jonringer) [inline pkgs/applications/misc/xneur/default.nix:28]:
```suggestion
      sha256 = "0pc17a4sdrnrc4z7gz28889b9ywqsm5mzm6m41h67j2f5zh9k3fy";
```
fetchpatch will do additional normalization

Comment (jtojnar) [reply pkgs/applications/misc/xneur/default.nix:28]: Or, as this is a static file, you can also just use `fetchurl`.

Comment (jnetod) [reply pkgs/applications/misc/xneur/default.nix]: Done

**Pull request merged**